crypto/x509/x_all.c

```diff
@@ -43,6 +43,11 @@ int X509_verify(X509 *a, EVP_PKEY *r)
 int X509_REQ_verify_ex(X509_REQ *a, EVP_PKEY *r, OSSL_LIB_CTX *libctx,
                        const char *propq)
 {
+    if (X509_REQ_get_version(a) != X509_REQ_VERSION_1) {
+        ERR_raise_data(ERR_LIB_X509, ERR_R_UNSUPPORTED, "version != 1");
```

Comment:
IMO `ERR_raise(ERR_LIB_X509, X509_R_UNSUPPORTED_VERSION);` would be more appropriate. Of course that requires running `make update` to add the new error reason.

Reply:
@t8m I fully agree. I avoided adding a new error reason to keep this pull request minimal and its impact easily observable.

However, since the library lacks version checks in most of the X509 code, I would recommend introducing this new error code along with stricter version checking for *all* X509 functions.